gormschema/gorm.go

```diff
@@ -187,3 +224,56 @@ func WithJoinTable(model any, field string, jointable any) Option {
 		})
 	}
 }
+
+func indirect(t reflect.Type) reflect.Type {
+	for t.Kind() == reflect.Ptr {
+		t = t.Elem()
+	}
+	return t
+}
+
+type (
+	// ViewOption is a configuration option for creating a viewBuilder.
+	ViewOption interface {
+		apply(*viewBuilder)
+	}
+	// ViewOptionFunc is an adapter to allow the use of functions as ViewOption.
+	ViewOptionFunc func(*viewBuilder)
```

Comment:
```suggestion
	// ViewOption is an adapter to allow the use of functions as ViewOption.
	ViewOption func(*viewBuilder)
```

I don't think the interface wrapping is needed